### PR TITLE
docs: fix developer guide commands

### PR DESCRIPTION
**Pull request by leewei05**
- fix make commands to match existing makefile
- add additional description for Python users

### CODE REVIEW

File: docs/contributing-devguide.md [hunk 1]
```diff
@@ -77,6 +77,13 @@ Run the following command to run a complete build. The rendered assets are avail
 ```shell
 make docs
 
+# Python 3
```
Comment (knelasevero) [inline docs/contributing-devguide.md:80]: I think we can `make serve-docs` here, right?

Comment (leewei05) [reply docs/contributing-devguide.md:80]: Yes, after testing the make commands. We need to `make docs` first, then we can `make serve-docs`.
I'll fix the commands in the guide, thanks for pointing it out. 😁

Comment (leewei05) [reply docs/contributing-devguide.md:80]: Since we can start our website with `make serve-docs`, should I just remove the python simple server?

Comment (knelasevero) [reply docs/contributing-devguide.md:80]: Makes sense to remove them. Thanks @leewei05 !

Comment (leewei05) [reply docs/contributing-devguide.md]: Just updated the fix, you're welcome. 😁

**Pull request merged**